### ISSUE TITLE
Smoke & CO detector not connecting to ring.com

### ISSUE THREAD

**Issue by 130s** [kinu-garage/hut_10sqft#624]
Cont'd from https://github.com/130s/30y-130s_life/issues/1030 about technical info of connecting First Alert's Z-wave sensor to ring.com. As in https://github.com/130s/30y-130s_life/issues/1030#issuecomment-984109411, the manufacturer agrees to replace the device I'm seeing an issue with, but I went ahead testing 2nd and 3rd devices I have and untested yet.

CoS
- [ ] 1st device hooked
- [x] 2nd device hooked
- [x] 3rd device hooked

**Comment by 130s** [comment]
- Tested out 2nd smoke detector. After struggle it finally connected to ring.com without ring complaining.
   - At first attempt I saw ring.com detected it but then failured after long time and said something but I don't remember. Since then, factory-reset this device (*1) and re-do on ring.com but ring doesn't detect it.
   - I've also tried the other option on ring that does NOT say "2nd get". This went well but then Ring doesn't detect smoke & CO detector, instead it detected a binary sensor without showing any particular info for smoke & CO sensor. So I'm not sure if ring connected to this smoke & CO, or happened to grab a random binary sensor.
      - And this binary sensor itself shows "`Critical Configuration Error`" even though the detection completed.
   - Well...then after some more trials, it finally connected. The atttempt made right before the successful detection still failed and suggested retry. This retry is the one that succeeded.
- 1st device is still failing to connect to ring.com.
   - I removed it from ring then just to verify its Z-wave capability, I tried to connect this device to another Z-wave gateway I have that uses ZwaveJS-MQTT https://github.com/130s/hut_10sqft/issues/610. It went well, and Home Assistant was able to monitor the smoke and CO sensor's status realtime. It lets me more visibility about the sensors.
      - So I can technically switch to HoA for smoke & CO sensing. However, thing pulling me off is that my HoA is not linked to professional monitoring service so I cannot get support in case of fire detected by these sensors (Related https://github.com/130s/hut_10sqft/issues/623).
   - Then after that, I did factory-reset *1 then let ring.com detect it and it says "Wrong Network". So clearly the sensor keeps info of old z-wave network. My way of Factory-rest *1 doesn't seem to be working here.
- Testing 3rd device.
   - Same as the 2nd device, first attempt failed, then ring.com app automatically retried and succeeded.
   - To prove steps are correct and reproduce success, I removed this device from ring.com, and did factory-reset itself (*1). Then attempted again to connect to ring.com

In summary,
- 2 out of 3 are now functioning with ring.com.
- 1 is waiting for a new usecase, or maybe replace with a new one as the manufacturer agrees.
- I think I'm just hitting flakiness of Z-wave init operation hopefully. Don't give up when an issue occurred once, keep trying and it often works out :/

*1...Instruction isn't clear at all. I found sliding the battery cartridge out, remove batteries and put them back in, press and hold Test button, slid the battery cartridge in, them release Test button.

**Comment by 130s** [comment]
> - 1st device is still failing to connect to ring.com.

Potential work is suggested in https://community.home-assistant.io/t/nortek-gc-tbz48-z-wave-thermostat-disconnects-too-frequently-from-ha/365448/12?

**Comment by 130s** [comment]
I had HA and ZwaveJS Mqtt re-included the device in question, then verified that is was removed again (at least from the GUI of those `http://%IPADDR%:8091/control-panel`). Then factory-reset (open the battery cartridge, remove batteries, wait for 6+ secs, put the batteries back in, press Test button, slide back in the cartridge while I still press Test, wait for beep of 1 or 2 secs, release Test button after 10+ secs since I started pressing), had ring.com include the device but it still says "wrong network". So out of ideas. Will call tech support.

**Comment by 130s** [comment]
Called support https://github.com/130s/30y-130s_life/issues/1030#issuecomment-984109411. Unlike prev. time, this time they asked me to call ring.com. They did verify the factory-reset I'm doing is appropriate.
